core/src/Exceptions/CoreException.cs

```diff
@@ -0,0 +1,85 @@
+﻿//----------------------------------------------------------------------
+//
+// Copyright (c) Microsoft Corporation.
+// All rights reserved.
+//
+// This code is licensed under the MIT License.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files(the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions :
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+//
+//------------------------------------------------------------------------------
+
+using System;
+using System.Globalization;
+
+namespace Microsoft.Identity.Core.Exceptions
+{
+    /// <summary>
+    ///     Abstract base exception, never thrown by itself.
+    /// </summary>
+    public abstract class CoreException : Exception

```

Comment:
do we want to expose the Core namespace? so far we have avoided doing that?
This probably means that customers won't be able to have ADAL.NET and MSAL.NET in the same application. We know that some customers do that (@bgavrilMS can elaborate)

Reply:
Customers should never see this exception directly since it's abstract and they'll still directly interact with MsalException or AdalException.  But OAuth2Client is in Core and we need to get to the ErrorCode value in order to fix this issue, so I had to get a common exception type in the base that we can check for.  Interested in advice/guidance to move this into a separate namespace or elsewhere since we can't make this an internal type because it is derived by public types.

Reply:
Too bad exceptions don't play nice with interfaces...